modules/light-clients/08-wasm/types/misbehaviour_handle_test.go

```diff
@@ -9,505 +9,100 @@ import (
-				clientState.UpdateState(suite.ctx, suite.chainA.Codec, suite.store, clientMsg)
+				suite.mockVM.RegisterQueryCallback(types.VerifyClientMessageMsg{}, func(_ wasmvm.Checksum, _ wasmvmtypes.Env, queryMsg []byte, store wasmvm.KVStore, _ wasmvm.GoAPI, _ wasmvm.Querier, _ wasmvm.GasMeter, _ uint64, _ wasmvmtypes.UFraction) ([]byte, uint64, error) {
+					var msg *types.QueryMsg
 
-				// Reset client state to the previous for the test
-				suite.chainA.App.GetIBCKeeper().ClientKeeper.SetClientState(suite.ctx, defaultWasmClientID, clientState)
+					err := json.Unmarshal(queryMsg, &msg)
+					suite.Require().NoError(err)
 
-				data, err = base64.StdEncoding.DecodeString(suite.testData["misbehaviour"])
-				suite.Require().NoError(err)
-				clientMsg = &types.ClientMessage{
-					Data: data,
-				}
+					suite.Require().NotNil(msg.VerifyClientMessage)
+					suite.Require().NotNil(msg.VerifyClientMessage.ClientMessage)
+					suite.Require().Nil(msg.Status)
+					suite.Require().Nil(msg.CheckForMisbehaviour)
+					suite.Require().Nil(msg.TimestampAtHeight)
+					suite.Require().Nil(msg.ExportMetadata)
+
+					resp, err := json.Marshal(types.EmptyResult{})
+					if err != nil {
+						return nil, 0, err
+					}
+
+					return resp, types.DefaultGasUsed, nil
+				})
 			},
-			true,
+			nil,
 		},
 		{
-			"trusted consensus state does not exist",
+			"failure: invalid client message",
 			func() {
-				data, err := base64.StdEncoding.DecodeString(suite.testData["misbehaviour"])
-				suite.Require().NoError(err)
-				clientMsg = &types.ClientMessage{
-					Data: data,
-				}
+				clientMsg = &ibctmtypes.Header{}
+
+				suite.mockVM.RegisterQueryCallback(types.VerifyClientMessageMsg{}, func(_ wasmvm.Checksum, _ wasmvmtypes.Env, queryMsg []byte, store wasmvm.KVStore, _ wasmvm.GoAPI, _ wasmvm.Querier, _ wasmvm.GasMeter, _ uint64, _ wasmvmtypes.UFraction) ([]byte, uint64, error) {
+					resp, err := json.Marshal(types.EmptyResult{})
+					suite.Require().NoError(err)
+
+					return resp, types.DefaultGasUsed, nil
+				})
 			},
-			false,
+			ibcerrors.ErrInvalidType,
 		},
 		{
-			"invalid wasm misbehaviour",
+			"failure: error return from contract vm",
 			func() {
-				clientMsg = &solomachine.Misbehaviour{}
+				suite.mockVM.RegisterQueryCallback(types.VerifyClientMessageMsg{}, func(_ wasmvm.Checksum, _ wasmvmtypes.Env, queryMsg []byte, store wasmvm.KVStore, _ wasmvm.GoAPI, _ wasmvm.Querier, _ wasmvm.GasMeter, _ uint64, _ wasmvmtypes.UFraction) ([]byte, uint64, error) {
+					return nil, 0, contractError
+				})
 			},
-			false,
+			contractError,
 		},
 	}
 
 	for _, tc := range testCases {
 		suite.Run(tc.name, func() {
-			suite.SetupWasmGrandpaWithChannel()
-			clientState, ok = suite.chainA.App.GetIBCKeeper().ClientKeeper.GetClientState(suite.ctx, defaultWasmClientID)
-			suite.Require().True(ok)
+			// reset suite to create fresh application state
+			suite.SetupWasmWithMockVM()
+			endpoint := wasmtesting.NewWasmEndpoint(suite.chainA)
```

Comment:
inb4 damian spacing comment

```suggestion
			suite.SetupWasmWithMockVM()

			endpoint := wasmtesting.NewWasmEndpoint(suite.chainA)
```